File: python/01-commands/tests/ComplexMethod/blacklist.py
import socket
import sys

blackListedIP = [
    "12.5.6.7",
    "192.0.5.1"
]

def get_current_ip_address():
    host_name = socket.gethostname()
    print(host_name)

    # Get the IP from gethostbyname method of socket module.
    return socket.gethostbyname(host_name)

def is_blacklisted(key):
    return key in blackListedIP

def is_whitelisted(key):
    return key == get_current_ip_address()

File: cpp/01-commands/tests/ComplexMethod/Blacklist.cpp
#include "pch.h"
#include "Blacklist.h"

/*static*/ 
std::vector<std::string> Blacklist::blackListedIP = { "12.5.6.7", "192.0.5.1" };

/*static*/
std::string Blacklist::GetCurrentIpAddress()
{
    return "10.5.0.14";
}

/*static*/
bool Blacklist::IsBlacklisted(std::string_view ip)
{
    return std::find(blackListedIP.begin(), blackListedIP.end(), ip) != blackListedIP.end();
}

/*static*/
bool Blacklist::IsWhiteList(std::string_view ip)
{
    return ip == GetCurrentIpAddress();
}